internal/resolver/resolver.go

```diff
@@ -578,13 +601,43 @@ func (r *resolver) loadAsFileOrDirectory(path string) (string, bool) {
 	return "", false
 }
 
+func resolvePathWithoutStar(from, path string) (string, error) {
+	replaced := strings.Replace(path, "/*", "", -1)
+	return fp.Join(from, replaced), nil
+}
+
 func (r *resolver) loadNodeModules(path string, dirInfo *dirInfo) (string, bool) {
 	for {
 		// Handle TypeScript base URLs for TypeScript code
-		if dirInfo.tsConfigJson != nil && dirInfo.tsConfigJson.absPathBaseUrl != nil {
-			basePath := r.fs.Join(*dirInfo.tsConfigJson.absPathBaseUrl, path)
-			if absolute, ok := r.loadAsFileOrDirectory(basePath); ok {
-				return absolute, true
+		if dirInfo.tsConfigJson != nil {
+
+			if dirInfo.tsConfigJson.absPathBaseUrl != nil {
+				if dirInfo.tsConfigJson.paths != nil {
+					for key, originalPaths := range dirInfo.tsConfigJson.paths {
+						for _, originalPath := range originalPaths {
+							if matched, err := regexp.MatchString("^"+key, path); matched && err == nil {
+								if absoluteOriginalPath, err := resolvePathWithoutStar(*dirInfo.tsConfigJson.absPathBaseUrl, originalPath); err == nil {
+									elements := strings.Split(path, "/")
+
+									elements = elements[1:]
+
+									resolved := append(strings.Split(absoluteOriginalPath, string(os.PathSeparator)), elements...)
+									basePath := strings.Join(resolved, "/")
+									if absolute, ok := r.loadAsFileOrDirectory(basePath); ok {
+										return absolute, true
+									}
+								}
+							}
+						}
+
+					}
+				} else {
+					basePath := r.fs.Join(*dirInfo.tsConfigJson.absPathBaseUrl, path)
```

Comment:
It looks like the TypeScript compiler still respects `baseUrl` if no `paths` matched, so we should do that too.

Reply:
thanks! resolved on 2587e08d326018a478aa1b0ec1c8b0837ad846a2